### PR TITLE
GQI - Logging in GQI extensions

### PR DESCRIPTION
**Pull request by WaHaeck**

### CODE REVIEW

File: user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/API_Reference/IGQILogger.md
Comment (skyline-rge) [inline user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/API_Reference/IGQILogger.md:1]: We'll need to mention that this is only available from DataMiner 10.4.5/10.5.0 onwards (with comment link to RN).
See other recent API ref additions for consistency.

File: user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/API_Reference/IGQILogger.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+---
+
+# IGQILogger interface
+
+## Definition
+
+- Namespace: `Skyline.DataMiner.Analytics.GenericInterface`
+- Assembly: `SLAnalyticsTypes.dll`
+
+Contains functionality to log messages and exceptions. The log files will be placed within the `C:\Skyline DataMiner\Logging\GQI\Extensions` folder.
```
Comment (skyline-rge) [inline user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/API_Reference/IGQILogger.md:12]: Update folder path

File: user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/API_Reference/IGQILogger.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+## Properties
+
+| Property | Type | Description |
+| -------- | ---- | ----------- |
+| MinimumLogLevel | GQILogLevel | Set or get the current minimum log level. The default value is `Information`. |
```
Comment (skyline-rge) [inline user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/API_Reference/IGQILogger.md:18]: We should make an API reference for the GQILogLevel as well, and link to it here.

Comment (skyline-rge) [inline user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/API_Reference/IGQILogger.md:18]:
```suggestion
| MinimumLogLevel | GQILogLevel | Get or set the minimum log level. The default is `GQILogLevel.Information`. |
```

File: user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/API_Reference/IGQILogger.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+GQILogLevel exposes the following log levels:
```
Comment (skyline-rge) [inline user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/API_Reference/IGQILogger.md:20]: This then belongs on the separate GQILogLevel page.

File: user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/API_Reference/IGQILogger.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+- **Debug**: Internal system events for debugging and troubleshooting.
+- **Information**: General operational messages to track application behavior.
+- **Warning** Indicates potential issues that may need attention.
+- **Error**: Signals unexpected failures or malfunctions in the application.
+- **Fatal**: Critical errors requiring immediate action, often leading to application termination.
+
+## Methods
+
+> [!NOTE]
+> The IGQILogger logger methods are inspired by [Serilog](https://github.com/serilog/serilog).
```
Comment (skyline-rge) [inline user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/API_Reference/IGQILogger.md:31]:
```suggestion
> The IGQILogger methods are inspired by [Serilog](https://github.com/serilog/serilog).
```

I'm wondering if we also need to mention that they actually use Serilog in the background and so the message will be interpreted as a Serilog template.

File: user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/API_Reference/OnInitInputArgs.md [hunk 5]
```diff
@@ -9,10 +9,11 @@ uid: GQI_OnInitInputArgs
...
 
-Contains functionality and general query information for user-definable data sources. An instance is provided through the `OnInit` life cycle method of the [IGQIOnInit building block](xref:GQI_IGQIOnInit).
+Contains functionality and general query information for ad hod data sources and custom operators. An instance is provided through the `OnInit` life cycle method of the [IGQIOnInit building block](xref:GQI_IGQIOnInit).
 
 ## Properties
 
 | Property | Type | Description |
 | -------- | ---- | ----------- |
 | DMS | [GQIDMS](xref:GQI_GQIDMS) | Object that can be used to interact with the DMS. |
+| Logger | [IGQILogger](xref:GQI_IGQILogger) | Object that can be used to log messages and exceptions. |
```
Comment (skyline-rge) [inline user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/API_Reference/OnInitInputArgs.md:19]: Need to mention that this in only available from a specific DataMiner version onwards.

File: user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/API_Reference/IGQILogger.md (hunk 4 above)
Comment (WaHaeck) [reply user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/API_Reference/IGQILogger.md:31]: Not the case anymore

File: user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/API_Reference/GQILogLevel.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+---
+
+# GQILogLevel enum
+
+## Definition
+
+- Namespace: `Skyline.DataMiner.Analytics.GenericInterface`
+- Assembly: `SLAnalyticsTypes.dll`
+
+Specifies constants that define the log level, used by [IGQILogger](xref:GQI_IGQILogger).
```
Comment (skyline-rge) [inline user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/API_Reference/GQILogLevel.md:12]: Remove comma

File: user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/API_Reference/IGQILogger.md [hunk 7]
```diff
@@ -32,48 +26,72 @@ GQILogLevel exposes the following log levels:
...
+Log a message with the [GQILogLevel.Error](xref:GQI_GQILogLevel) level.
+
+Should be used to record unexpected failures or malfunctions in the application, signaling issues that need to be addressed.
 
 ### void Fatal(string message)
 
-Logs critical errors that result in the termination of the application or loss of essential functionality, requiring immediate action to resolve.
+Log a message with the [GQILogLevel.Fatal](xref:GQI_GQILogLevel) level.
+
+Should be used to log critical errors that result in the termination of the application or loss of essential functionality, requiring immediate action to resolve.
```
Comment (skyline-rge) [inline user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/API_Reference/IGQILogger.md:91]: "termination of the application" is not really applicable in this context.

**Review by MariekeGO — APPROVED** [review]

**Pull request merged**